docs/docs/security.md

```diff
@@ -1,167 +1,162 @@
+data the enclave accesses. Other types of enclaves work purely with secret data but expect that the host
+isn't normally malicious. In this scenario, enclaves are used to slow down or stop attackers in the face of a hacked
+host network. It thus makes up one part of a standard suite of security measures.
 
-Because the performance/privacy tradeoff presented by side channel attacks can vary so widely, and this is an active area
-of academic research, the expectation is that every new Conclave version will provide new tools and tunable settings 
-to control their impact. This will continue for the lifespan of the product. As a developer it's your responsibility to
-both upgrade to new versions as they come out, and to take side channels into account when planning the architecture
-and capacity needs of your application.
+As the performance/privacy tradeoff presented by side-channel attacks can vary widely, and this is an active area
+of academic research, Conclave will provide new tools and tunable settings to control their impact with new versions.
+As a developer, you need to upgrade to new versions as they come out, and take other mitigations steps against 
+side-channel attacks when planning the architecture of your application.
 
 ### Random numbers
 
-Care must be taken when generating random numbers inside the enclave to ensure that you use a suitably strong
-random number generator that cannot be weakened or influenced by the host by tampering with data that is used
-to seed the random number generator.
-
-The safest way to ensure you are using a safe random number generator is to use the `SecureRandom` class to generate
-your random numbers. In Conclave enclaves, the implementation of this class uses the `RDRAND` instruction to 
-use an on-CPU hardware random number generator.
-
-The implementation of the JDK `Random` class normally uses the system time as a seed, XORing it with a fixed value
-that is updated predictably with each new `Random` class instance. This provides an attack surface for the
-host as the time inside the enclave is provided directly by the host. In order to prevent accidental use of
-an insecure random number generator, Conclave modifies this behaviour so the first instance of a `Random`
-class XORs the system time with a truly random number derived from the `RDRAND` instruction. This results in
-the class using a truly unique high entropy seed. Subsequent instances of `Random` transform the seed to
-ensure no two sequences are the same but the entropy is not renewed using the `RDRAND` instruction.
-Therefore for the best quality random data it is still recommended to use `SecureRandom`.
\ No newline at end of file
+To generate random numbers inside the enclave, you need to use a strong random number generator that cannot be
+weakened by the host.
+
+The safest way is to use the `SecureRandom` class to generate random numbers. In Conclave enclaves, the 
```

Comment:
Link the SecureRandom docs: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/SecureRandom.html

Reply:
Done.